### DISCUSSION TITLE
How can I copy a file absolute path to the system's clipboard?

### DISCUSSION THREAD

**Discussion by brielov** [kyoheiu/felix#278] [Q&A]
Suppose I'm browsing a directory filled with files with long and random names. How can I copy the currently highlighted file's path to the clipboard? So I can maybe paste it using `wl-clipboard`.

**Comment by kyoheiu** [comment]
@brielov 

Currently there is no keybinding to copy the full path of an item.
One way is `:readlink -f <file-name> | wlcopy`, which reads the full path and pipe it to the clipboard command.

You could also use a shell script in felix:
```
# .bashrc on macOS
# You should change pbcopy to wlcopy.
function fp() {
    readlink -f "$1" | pbcopy
}
export -f fp
```
And after re-launching the shell, `:fp <file-name>` in felix should work.
Note that you can use `<C-r>a` to paste item name to the command line.

```
<C-r>a         :In the command line, paste item name in register a.
```

Also note that it's hard to support the external clipboard program, see this issue https://github.com/kyoheiu/felix/issues/226 .
